TFM Issue Triage Documentation (#811)
Azure/Azure-Verified-Modules: docs/content/help-support/issue-triage/tfm-issue-triage.md

---

> title: TFM Issue Triage

Reviewer: ```suggestion
title: Terraform Issue Triage
```

geekdocNav: true
geekdocAlign: left
geekdocAnchor: true
geekdocToC: 1
---

{{< toc >}}

## Overview

> This page provides guidance for **Terraform Module (TFM) owners** on how to triage **AVM module issues** and **AVM question/feedback** items filed in their Terraform Module Repo, as well as how to manage these GitHub issues throughout their lifecycle.

Reviewer: ```suggestion
This page provides guidance for **Terraform Module owners** on how to triage **AVM module issues** and **AVM question/feedback** items filed in their Terraform Module Repo(s), as well as how to manage these GitHub issues throughout their lifecycle.
```

As such, **the following issues are to be filed in the TFM repository**:

- **AVM Module Issue**: Issues specifically related to an existing AVM module, such as feature requests, bug and security bug reports.
- **AVM Question/Feedback**:Generic feedback and questions, related to existing AVM module, the overall framework, or its automation (CI environment).

> Do **NOT** file the following types of issues in the **TFM repository**, as they **MUST** be tracked in the **AVM repo**:

Reviewer: ```suggestion
Do **NOT** file the following types of issues in a **Terraform repository**, as they **MUST** be tracked in the **AVM repo**:
```

> - **\[[Module Proposal](https://aka.ms/AVM/ModuleProposal)]**: Proposals for new AVM modules, or for migrating existing CARML/TFVM modules to AVM.

Reviewer: ```suggestion
- **\[[Module Proposal](https://aka.ms/AVM/ModuleProposal)]**: Proposals for new AVM modules.
```

- **\[[Orphaned Module](https://aka.ms/AVM/OrphanedModule)]**: Indicate that a module is orphaned (has no owner).
- **\[[Question/Feedback](https://aka.ms/AVM/QuestionFeedback)]**: Generic questions/requests related to the AVM site or documentation.

{{< hint type=note >}}
Every module needs a module proposal to be created in the AVM repository. This applies to both net new modules, as well as modules that are to be migrated from CARML/TFVM!
{{< /hint >}}

<br>

## Module Owner Responsibilities

During the triage process, module owners are expected to check, complete and follow up on the items described in the sections below.

Module owners **MUST** meet the SLAs defined on the [Module Support](/Azure-Verified-Modules/help-support/module-support/) page! While there's automation in place to support meeting these SLAs, module owners **MUST** check for new issues on a regular basis.
<!--
{{< hint type=important >}}
The [BRM repository](https://aka.ms/BRM) includes other, non-AVM modules and related GitHub issues. As a module owner, make sure you're only triaging, managing or otherwise working on issues that are related to AVM modules!
{{< /hint >}}

> -->

Reviewer 2: comment could be removed

{{< hint type=tip >}}

- To look for items that **need triaging**, look for issue labled with ➡️ <mark style="background-color:#FBCA04;">Needs: Triage 🔍</mark>⬅️.
- To look for items that **need attention**, look for issue labled with ➡️ <mark style="background-color:#E99695;color:white;">Needs: Attention 👋</mark> ⬅️.

{{< /hint >}}

<br>

## Module Issue

An issue is considered to be an "AVM module issue" if

> - it was opened through the **AVM Module Issue** template in the TFM repository,

Reviewer: ```suggestion
- it was opened through the **AVM Module Issue** template in the Terraform repository,
```

    - Make sure the module's details are captured correctly in the description - i.e., name, classification (resource/pattern), language (Bicep/Terraform), etc.
    - Make sure the issue is categorized using one of the following type labels:
      - "<mark style="background-color:#A2EEEF;">Type: Feature Request ➕</mark>"
      - "<mark style="background-color:#D73A4A;color:white;">Type: Bug 🐛</mark>"
      - "<mark style="background-color:#FFFF00;">Type: Security Bug 🔒</mark>"
2. Apply relevant labels for module classification (resource/pattern): "<mark style="background-color:#D3D3D3;">Class: Resource Module 📦</mark>" or "<mark style="background-color:#A9A9A9;">Class: Pattern Module 📦</mark>"
3. Communicate next steps to the requestor (issue author).
4. Remove the "<mark style="background-color:#FBCA04;">Needs: Triage 🔍</mark>" label.
5. When more detailed plans are available, communicate expected timeline for the update/fix to the requestor (issue author).
6. Only close the issue, once the next version of the module was fully developed, tested and published.

<br>

## General Question/Feedback and other standard issues

An issue is considered to be an "AVM Question/Feedback" if

> - it was opened through the **AVM Question/Feedback** template in your TFM repository,

Reviewer: ```suggestion
- it was opened through the **AVM Question/Feedback** template in your Terraform repository,
```
